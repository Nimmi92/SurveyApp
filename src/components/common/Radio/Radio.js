import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { device } from '../../../device/device';


const Radio = (props) => {
 const {data} = props;
  return (
    <div>
     { data.options && data.options.map((o,i) => {
      return <div><input type="radio" name={data.name} value={o} /><label>{o}</label></div>
    })
   }
     
    </div>
  );
};


const { string, array } = PropTypes;

Radio.propTypes = {
	options: array.isRequired,
	name: string.isRequired
}

export default Radio;
