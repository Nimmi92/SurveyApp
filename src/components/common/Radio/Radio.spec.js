import React from 'react';
import { shallow } from 'enzyme';
import Radio from './Radio';
import toJson from "enzyme-to-json";

describe('<Radio />', () => {
  it('should be an radio button', () => {
    const props = {
      name: 'gender',
      options: ['select','one','two'],
      onChange: jest.fn()
    };

    const wrapper = shallow(<Radio {...props} onChange={props.onChange}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

})