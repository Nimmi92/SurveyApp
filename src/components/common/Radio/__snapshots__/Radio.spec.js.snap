// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> should be an radio button 1`] = `
<ForwardRef>
  <div
    key="0"
  >
    <input
      name="gender"
      onChange={[MockFunction]}
      type="radio"
      value="select"
    />
    <label>
      select
    </label>
  </div>
  <div
    key="1"
  >
    <input
      name="gender"
      onChange={[MockFunction]}
      type="radio"
      value="one"
    />
    <label>
      one
    </label>
  </div>
  <div
    key="2"
  >
    <input
      name="gender"
      onChange={[MockFunction]}
      type="radio"
      value="two"
    />
    <label>
      two
    </label>
  </div>
</ForwardRef>
`;
